resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -51,33 +52,16 @@ import org.apache.spark.util._
 /**
  * Common application master functionality for Spark on Yarn.
  */
-private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends Logging {
+private[spark] class ApplicationMaster(args: ApplicationMasterArguments, sparkConf: SparkConf,
```

Comment:
This doesn't follow Spark's convention for multi-line arguments.

This also looks a little odd now, because there are conflicting arguments. `ApplicationMasterArguments` is now only used in cluster mode, and everything else is expected to be provided in the other parameters. So while this is the simpler change, it's also a little ugly.

I don't really have a good suggestion right now, but it's something to think about.

Reply:
I made changes to the default constructor and added another constructor. Please check and let me know anything can be done better.